src/frontend/Semantic_check.ml

```diff
@@ -290,9 +290,45 @@ let semantic_check_fn_normal ~is_cond_dist ~loc id es =
     | Some _ ->
         (* Check that Funaps are actually functions *)
         Semantic_error.returning_fn_expected_nonfn_found loc id.name |> error
-    | None ->
-        Semantic_error.returning_fn_expected_undeclaredident_found loc id.name
-        |> error)
+    | None -> (
+        let gen_prefix l = String.concat ~sep:"_" (List.rev l) in
+        match List.rev (String.split id.name ~on:'_') with
+        | (("lpmf" | "lupmf") as suffix) :: tl
```

Comment:
Following #995 (which I'm hoping gets merged today) I think this can be made simpler with this:

https://github.com/WardBrian/stanc3/blob/31e04c11a9b84c42955656d477849f7e2fff3741/src/middle/Utils.ml#L22-L28

I think this also avoids the need for the new lists in `Utils`

Reply:
OK, will wait!